src/frontend/src/optimizer/plan_node/logical_hop_window.rs

```diff
@@ -121,8 +138,36 @@ impl fmt::Display for LogicalHopWindow {
 }
 
 impl ColPrunable for LogicalHopWindow {
-    fn prune_col(&self, _required_cols: &FixedBitSet) -> PlanRef {
-        unimplemented!("LogicalHopWindow::prune_col")
+    fn prune_col(&self, required_cols: &FixedBitSet) -> PlanRef {
+        self.must_contain_columns(required_cols);
+        let require_win_start = required_cols.contains(self.window_start_col_idx());
+        let require_win_end = required_cols.contains(self.window_end_col_idx());
+
+        let o2i = self.o2i_col_mapping();
+        let input_required_cols = o2i.rewrite_bitset(required_cols);
+        let input = self.input.prune_col(&input_required_cols);
+        let input_change = ColIndexMapping::with_remaining_columns(&input_required_cols);
+        let (new_hop, _) = self.rewrite_with_input(input, input_change);
+        match (require_win_start, require_win_end) {
+            (true, true) => new_hop.into(),
+            (true, false) => {
+                let proj_map = ColIndexMapping::identity_or_none(
+                    new_hop.schema().len(),
+                    new_hop.schema().len() - 1,
+                );
+                LogicalProject::with_mapping(new_hop.into(), proj_map)
+            }
+            (false, true) => {
+                let mut proj_map = ColIndexMapping::identity_or_none(
+                    new_hop.schema().len(),
+                    new_hop.schema().len() - 1,
+                );
+                proj_map.put(new_hop.schema().len() - 1, Some(new_hop.schema().len() - 2));
+                proj_map.put(new_hop.schema().len() - 2, None);
+                LogicalProject::with_mapping(new_hop.into(), proj_map)
+            }
+            (false, false) => unreachable!(),
```

Comment:
IMO it's a valid SQL that user use HOP but don't really use `window_start` and `window_end`